docs/src/design-system/building-blocks/button/ButtonStatesExample.tsx

```diff
@@ -1,107 +1,127 @@
-import { Box } from '@mui/material';
 import {
+  Box,
   Table,
   TableBody,
   TableCell,
   tableCellClasses,
   TableContainer,
   TableHead,
   TableRow,
+  Theme,
   ThemeProvider,
   useTheme,
 } from '@mui/material';
 import { Button, Typography } from '@ui-kit-2022/components';
-import { dark } from '@ui-kit-2022/theme';
 import React from 'react';
 
-interface Props {
+export interface Props {
   variantType?: 'PRIMARY' | 'SECONDARY' | 'TERTIARY';
+  theme: Theme;
 }
-const DarkBgState: React.FC<Props> = ({ variantType }) => {
-  const { palette } = useTheme();
+
+const ButtonStatesExample: React.FC<Props> = ({ variantType, theme = useTheme() }) => {
+  const bgColor = (theme: Theme) => {
+    return {
+      backgroundColor: {
+        light: theme.palette.grey[50],
+        dark: theme.palette.background.paper,
+      }[theme.palette.mode],
+    };
+  };
```

Comment:
This can be simpler:

```
const backgroundColor = {
    light: theme.palette.grey[50],
    dark: theme.palette.background.paper,
}[theme.palette.mode];
```

or use a ternary (more conventional)

`const backgroundColor = theme.palette.mode === 'light' ? ... : ... ;`

The `sx` props can be simplified when `backgroundColor` has the same property name and variable name:

`sx={{ backgroundColor }}`

or as needed

```
sx={{
    backgroundColor,
    borderTopRadius: ...,
    ...
}}
```

Reply:
Much simpler than I did !